nova/tests/unit/scheduler/weights/test_weights_resize_same_shard.py

```diff
@@ -0,0 +1,106 @@
+from nova.tests.unit.scheduler import fakes
+from nova.tests import uuidsentinel as uuids
+
+
+class PreferSameShardOnResizeWeigherTestCase(test.NoDBTestCase):
+    def setUp(self):
+        super(PreferSameShardOnResizeWeigherTestCase, self).setUp()
+        self.weight_handler = weights.HostWeightHandler()
+        self.weighers = [same_shard.PreferSameShardOnResizeWeigher()]
+
+        self.instance1 = objects.Instance(host='host1', uuid=uuids.instance_1)
+        self.instance2 = objects.Instance(host='host1', uuid=uuids.instance_2)
+        self.instance3 = objects.Instance(host='host2', uuid=uuids.instance_3)
+        self.aggs1 = [objects.Aggregate(id=1, name='az-a', hosts=['host1']),
+                      objects.Aggregate(id=1, name='vc-a-0', hosts=['host1'])]
+        self.aggs2 = [objects.Aggregate(id=2, name='az-b', hosts=['host2']),
+                      objects.Aggregate(id=2, name='vc-a-1', hosts=['host2'])]
+
+        self.hosts = [
+            fakes.FakeHostState('host1', 'compute', {'aggregates': self.aggs1,
+                'instances': {
+                    self.instance1.uuid: self.instance1,
+                    self.instance2.uuid: self.instance2,
+                }}),
+            fakes.FakeHostState('host2', 'compute', {'aggregates': self.aggs2,
+                'instances': {
+                    self.instance3.uuid: self.instance3,
+                }}),
+        ]
+
+    def test_prefer_resize_to_same_shard(self):
+        self.flags(prefer_same_shard_resize_weight_multiplier=1.0,
+                   group='filter_scheduler')
+        request_spec = objects.RequestSpec(
+            instance_uuid=self.instance1.uuid,
+            scheduler_hints={'_nova_check_type': ['resize']})
+        weighed_hosts = self.weight_handler.get_weighed_objects(
+            self.weighers, self.hosts, request_spec)
+        self.assertEqual(1.0, weighed_hosts[0].weight)
+        self.assertEqual(0.0, weighed_hosts[1].weight)
+        self.assertEqual('host1', weighed_hosts[0].obj.host)
+
+    def test_prefer_resize_to_different_aggregate(self):
+        self.flags(prefer_same_shard_resize_weight_multiplier=-1.0,
+                   group='filter_scheduler')
+        request_spec = objects.RequestSpec(
+            instance_uuid=self.instance1.uuid,
+            scheduler_hints={'_nova_check_type': ['resize']})
+        weighed_hosts = self.weight_handler.get_weighed_objects(
+            self.weighers, self.hosts, request_spec)
+        self.assertEqual(0.0, weighed_hosts[0].weight)
+        self.assertEqual(-1.0, weighed_hosts[1].weight)
+        self.assertEqual('host2', weighed_hosts[0].obj.host)
+
+    def test_ignore_scheduling_new_instance(self):
+        self.flags(prefer_same_shard_resize_weight_multiplier=1.0,
+                   group='filter_scheduler')
+        request_spec = objects.RequestSpec(
+            instance_uuid=uuids.fake_new_instance,
+            scheduler_hints={'_nova_check_type': ['resize']})
```

Comment:
Why does a new instance have `_nova_check_type: ['resize']` set?

Reply:
just to tell that even if resize is triggered with new instance it wont work since instance is new and "source_host" is missing

Reply:
This is not clear from the code, as a new instance normally wouldn't have `resize` set. Please add a comment that explains why, so anybody reading the code later on can understand it.